models/turbine_models/custom_models/sdxl_inference/unet.py

```diff
@@ -67,12 +74,85 @@ def forward(
             added_cond_kwargs=added_cond_kwargs,
             return_dict=False,
         )[0]
+        if self.do_classifier_free_guidance:
+            noise_preds = noise_pred.chunk(2)
+            noise_pred = noise_preds[0] + guidance_scale * (
+                noise_preds[1] - noise_preds[0]
+            )
         return noise_pred
 
 
+def get_punet_model(hf_model_name, external_weight_path, precision="i8"):
+    from sharktank.models.punet.model import (
+        Unet2DConditionModel as sharktank_unet2d,
+        ClassifierFreeGuidanceUnetModel as sharktank_CFGPunetModel,
+    )
+    from sharktank.utils import cli
+
+    if precision == "i8":
+        repo_id = "amd-shark/sdxl-quant-models"
+        subfolder = "unet/int8"
+        revision = "82e06d6ea22ac78102a9aded69e8ddfb9fa4ae37"
+    elif precision in ["fp16", "fp32"]:
```

Comment:
FYI - the sdxl-quant-models do have an fp16 variant IRPA file. I use that for various comparisons, etc.

https://huggingface.co/amd-shark/sdxl-quant-models/tree/main/unet/fp16/export

Reply:
I guess we can decide whether we want to do the full model download/quantization here or just fetch the IRPA file from the export subdir. If we did the latter, you could drop get_punet_dataset() that is actually building it and just use the one we publish. If doing that, you can eliminate this branch and just change the subfolder and filename or something.

Fine to check in like this since it is working, but can simplify by leaning more on published artifacts.

Now that I'm thinking of it, though, we're going to need a procedure for ML-Perf to rely completely on generated artifacts. Currently, that is being done by providing scripts to regenerate, but then we probably also need to support flags here and elsewhere to side-load.

In any case, let's land and discuss/iterate.